### PR TITLE
Unset the current-context after minikube stop

### PR DESCRIPTION
**Pull request by hpandeycodeit**
This PR implements Feature Request #4154

### CODE REVIEW

File: pkg/util/kubeconfig.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func PopulateKubeConfig(cfg *KubeConfigSetup, kubecfg *api.Config) error {
 	// Only set current context to minikube if the user has not used the keepContext flag
 	if !cfg.KeepContext {
 		kubecfg.CurrentContext = cfg.ClusterName
+		//kubecfg.CurrentContext = ""
```
Comment (tstromberg) [inline pkg/util/kubeconfig.go]: Remove?

**Comment by tstromberg** [comment]
@minikube-bot OK to test

File: pkg/util/kubeconfig.go [hunk 2]
```diff
@@ -312,3 +313,16 @@ func GetPortFromKubeConfig(filename, machineName string) (int, error) {
 	port, err := strconv.Atoi(kport)
 	return port, err
 }
+
+//UnsetCurrentContext unsets the current-context from minikube to "" on minikube stop
+func UnsetCurrentContext(filename, machineName string)  error {
```
Comment (tstromberg) [inline pkg/util/kubeconfig.go]: Please run `gofmt -s -w` on this file to reformat it to Go standards. Consider configuring your code editor to do this on save.

File: pkg/util/kubeconfig.go (hunk 1 above)
Comment (hpandeycodeit) [reply pkg/util/kubeconfig.go]: done

File: pkg/util/kubeconfig.go (hunk 2 above)
Comment (hpandeycodeit) [reply pkg/util/kubeconfig.go]: formatted this one.

**Comment by hpandeycodeit** [comment]
@tstromberg I have included the tests. Let me know if it looks okay. Thanks for the pointer :)

**Comment by tstromberg** [comment]
Looks great, but you'll need to do a tiny bit more work to plumb the runner through correctly:

```
test/integration/start_stop_delete_test.go:81:4: undefined: kubeRunner
test/integration/start_stop_delete_test.go:81:17: undefined: "k8s.io/minikube/test/integration/util".NewKubeCtlRunner
test/integration/start_stop_delete_test.go:82:4: undefined: currentContext
test/integration/start_stop_delete_test.go:82:21: undefined: kubeRunner
test/integration/start_stop_delete_test.go:83:7: undefined: currentContext
test/integration/start_stop_delete_test.go:92:4: undefined: currentContext
test/integration/start_stop_delete_test.go:92:21: undefined: kubeRunner
test/integration/start_stop_delete_test.go:93:7: undefined: currentContext
```

Also, please run `gofmt -s -w` on the files you changed so that they are formatted with the recommended Go style.  FYI, you can run the integration tests locally by using something like:

```
make test
env TEST_ARGS="-minikube-start-args=--vm-driver=kvm2 -test.run TestStartStop" make integration
```

**Comment by hpandeycodeit** [comment]
@tstromberg Fixed the gofmt errors. Integration tests passed as well. Hopefully it should be fine this time.

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
Tests are failing:

```
--- FAIL: TestStartStop (812.27s)
    --- FAIL: TestStartStop/nocache_oldest (166.82s)
        start_stop_delete_test.go:87: current-context not set to minikube
    --- FAIL: TestStartStop/feature_gates_newest_cni (199.59s)
        start_stop_delete_test.go:87: current-context not set to minikube
    --- FAIL: TestStartStop/containerd_and_non_default_apiserver_port (224.69s)
        start_stop_delete_test.go:87: current-context not set to minikube
    --- FAIL: TestStartStop/crio_ignore_preflights (221.17s)
        start_stop_delete_test.go:87: current-context not set to minikube
 ```

**Comment by tstromberg** [comment]
It might help here if the test showed the error in got vs want syntax, such as:

got context %q, want context %q

**Comment by hpandeycodeit** [comment]
@tstromberg I have changed it. Not sure why the context is different. Not getting this error in my local though.

**Comment by hpandeycodeit** [comment]
@tstromberg thanks for approving the changes! 

but above failing tests still showing a failure in current-context mismatch. 

`
--- FAIL: TestStartStop (826.16s)
    --- FAIL: TestStartStop/nocache_oldest (179.80s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/feature_gates_newest_cni (203.97s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/containerd_and_non_default_apiserver_port (222.13s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/crio_ignore_preflights (220.26s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
=== RUN   TestVersionUpgrade
`

not sure from where it's getting "minikube\n" 

Any Idea?

**Comment by tstromberg** [comment]
```
--- FAIL: TestStartStop (947.48s)
    --- FAIL: TestStartStop/nocache_oldest (210.63s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/feature_gates_newest_cni (223.77s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/containerd_and_non_default_apiserver_port (252.40s)
        start_stop_delete_test.go:87: got current-context - "minikube\n", want  current-context "minikube"
    --- FAIL: TestStartStop/crio_ignore_preflights (260.68s)
```

You may want to call strings.TrimRight() before comparing the value.

**Comment by tstromberg** [comment]
Looks like it's getting better, but something is still not quite right:

```
--- FAIL: TestStartStop (857.56s)
    --- FAIL: TestStartStop/nocache_oldest (186.06s)
        start_stop_delete_test.go:100: Failed to unset the current-context "minikube\n"
    --- FAIL: TestStartStop/feature_gates_newest_cni (206.81s)
        start_stop_delete_test.go:100: Failed to unset the current-context "minikube\n"
    --- FAIL: TestStartStop/containerd_and_non_default_apiserver_port (238.16s)
        start_stop_delete_test.go:100: Failed to unset the current-context "minikube\n"
    --- FAIL: TestStartStop/crio_ignore_preflights (226.53s)
        start_stop_delete_test.go:100: Failed to unset the current-context "minikube\n"
```

Does running the integration test work for you?

**Comment by hpandeycodeit** [comment]
@tstromberg I am able to test this in my local and it's failing as well. So the issue here is, if I put this code right below the stop, since the minikube takes time to stop and the code still retrieves current-context as "minikube": 

```
checkStop := func() error {
	r.RunCommand("stop", true)
	return r.CheckStatusNoFail(state.Stopped.String())
}

currentContext, err = kubectlRunner.RunCommand([]string{"config", "current-context"})
if strings.TrimRight(string(currentContext), "\n") != "" {
	t.Fatalf("Failed to unset the current-context %q", strings.TrimRight(string(currentContext), "\n"))
}

if err := util.Retry(t, checkStop, 5*time.Second, 6); err != nil {
	t.Fatalf("timed out while checking stopped status: %v", err)
}


```

However if I put this code after the code that checks stopped status, minikube is stopped then but there is no context and then it throws error something like this 

**util.go:295: Temporary Error: error running command [config current-context]: exit status 1. Stdout: 
error: current-context is not set**

So looks like I can't really compare the current-context with an empty string here ""

**Comment by hpandeycodeit** [comment]
@tstromberg let me know if this looks okay to you or any other change is needed to the test.

**Comment by tstromberg** [comment]
Looks good. Thank you!

**Comment by hpandeycodeit** [comment]
Thank you @tstromberg!

**Pull request merged**